tests/TypeInfer.tables.test.cpp

```diff
@@ -3642,4 +3642,34 @@ end
     LUAU_REQUIRE_NO_ERRORS(result);
 }
 
+TEST_CASE_FIXTURE(BuiltinsFixture, "index_property_table_intersection_1")
```

Comment:
Optionally, this can be moved to `TypeInfer.intersectionTypes.test.cpp`, that already contains tests similar to this one, but obviously, not the _2 version.